File: frontend/src/api/dto/thumbnailRegenerationResponse.ts
import type { ThumbnailOwnerReport } from './thumbnailOwnerReport.ts'

/**
 * Outcome of a thumbnail regeneration, one entry per entity
 */
export interface ThumbnailRegenerationResponse {
  /** Whether this was a dry run (nothing redrawn) */
  dryRun: boolean
  /** Number of entities matching the criteria, before the limit */
  matched: number
  /** Entities processed, at most the limit */
  entities: ThumbnailOwnerReport[]
}

File: backend/src/main/java/fr/pedalons/dto/admin/ThumbnailRegenerationResponse.java
package fr.pedalons.dto.admin;

import fr.pedalons.dto.validation.ValidateSchema;
import java.util.List;
import org.eclipse.microprofile.openapi.annotations.media.Schema;

@Schema(description = "Outcome of a thumbnail regeneration, one entry per entity")
@ValidateSchema
public record ThumbnailRegenerationResponse(
    @Schema(description = "Whether this was a dry run (nothing redrawn)", required = true)
        boolean dryRun,
    @Schema(
            description = "Number of entities matching the criteria, before the limit",
            required = true)
        int matched,
    @Schema(description = "Entities processed, at most the limit", required = true)
        List<ThumbnailOwnerReport> entities) {}
